configure.ac

```diff
@@ -4147,9 +4147,7 @@ AS_VAR_IF([ac_cv_header_gdbm_dash_ndbm_h], [yes], [
 AS_UNSET([ac_cv_header_gdbm_ndbm_h])
 
 if test "$ac_cv_header_gdbm_slash_ndbm_h" = yes -o "$ac_cv_header_gdbm_dash_ndbm_h" = yes; then
-  WITH_SAVE_ENV([
-    AC_SEARCH_LIBS([dbm_open], [gdbm_compat])
-  ])
+  AC_CHECK_LIB(gdbm_compat, dbm_open, [have_gdbm_compat=yes], [have_gdbm_compat=no])
```

Comment:
Why not just add these actions to the existing `AC_SEARCH_LIBS`?

Reply:
`AC_SEARCH_LIBS` caches results, so if the previous invocation found `-lndbm`, the repeated call would also use `-lndbm` even though it's not passed to the second argument.

In other words, calling `AC_SEARCH_LIBS(a, b)` with the same `a` but different `b` is invalid.

Reply:
Yes, we'd need to `AS_UNSET([ac_cv_search_dbm_open])`.

Reply:
That sounds like playing with implementation details, and I don't really understand what we gain from that.

Reply:
It is, but it is also how we handle most of the dbm detection (there are three `AS_UNSET`s there already)[^1]. There is value in keeping the diff as small as possible.

How about this: we make this (small) change, and then we look at how we can improve the existing xdbm dependency detection code in a series of AC refactorings and improvements.

[^1]: I know that the existing code might not be best practice AC code